src/Microsoft.Sbom.Extensions.DependencyInjection/MSBuildLogger.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Sbom.Extensions.DependencyInjection;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Serilog.Events;
+using ILogger = Serilog.ILogger;
+
+/// <summary>
+/// A class to remap Errors logged from ComponentDetection assemblies to Warnings.
+/// </summary>
+public class MSBuildLogger : ILogger
+{
+    private readonly TaskLoggingHelper loggingHelper;
+
+    public MSBuildLogger(TaskLoggingHelper loggingHelperToWrap)
+    {
+        loggingHelper = loggingHelperToWrap;
+    }
+
+    public void Write(LogEvent logEvent)
+    {
+        var logLevel = logEvent.Level;
+        switch (logLevel)
+        {
+            case LogEventLevel.Debug:
+                loggingHelper.LogMessage(MessageImportance.Low, logEvent.RenderMessage());
+                break;
+            case LogEventLevel.Information:
+                loggingHelper.LogMessage(MessageImportance.High, logEvent.RenderMessage());
+                break;
+            case LogEventLevel.Warning:
+                loggingHelper.LogWarning(logEvent.RenderMessage());
+                break;
+            case LogEventLevel.Error:
+            case LogEventLevel.Fatal:
+                loggingHelper.LogError(logEvent.RenderMessage());
+                break;
+            default:
+                break;
+        }
```

Comment:
This is the core of the Serilog<->MSBuild bridge

Reply:
As called out elsewhere, I'd love to see this live in the `Microsoft.Sbom.Targets` project, and use some sort of delegate to pass it into the DependencyInjection project. The idea is to keep the MDBuild dependencies tightly scoped, since their functionality is exclusive to the NuGet targeting